### PR TITLE
interval-str needs to display no value when large milliseconds value is passed in

### PR DESCRIPTION
**Pull request by sc00terUL**
Change the display of minutes and seconds to use str instead of format as well.

### CODE REVIEW

**Comment by weavejester** [comment]
Thanks for the PR, but unfortunately this changes the display; `(str x ":" y)` is not the same as `(format "%02d:%02d" x y)`. As I mentioned in issue #5, sufficiently large intervals (say, > 1 year) should result in a `"--:--"` string, as there's no point in giving the user nonsensically large numbers.

File: src/progrock/core.clj [hunk 1]
```diff
@@ -42,11 +42,12 @@
   (if (pos? total) (int (* 100 (/ progress total))) 0))
 
 (defn- interval-str [milliseconds]
-  (if (nil? milliseconds)
-    "--:--"
-    (let [seconds (mod (int (/ milliseconds 1000)) 60)
-          minutes (int (/ milliseconds 60000))]
-      (format "%02d:%02d" minutes seconds))))
+  (if (and (some? milliseconds)
+           (> Integer/MAX_VALUE milliseconds))
```
Comment (weavejester) [inline src/progrock/core.clj]: What about:

```clojure
(if (or (nil? milliseconds)
        (> milliseconds Integer/MAX_VALUE))
  "--:--"
```

That way it's changing less of the code.

**Comment by weavejester** [comment]
Thanks! That looks good. You just need to fix the indentation on line 46, remove the threading change as that's unnecessary, then squash the commits and give them a suitable commit message like:

```
Check integer max value when displaying times

Fixes #5.
```

I can also make the change if you want. Just let me know.

**Comment by sc00terUL** [comment]
I fixed the indentation but I like the threading as it makes things easier to read for me.

I'll leave it to you to squash commit.

**Comment by weavejester** [comment]
Okay, I'll squash your commits and remove the threading changes. The threading might be nicer to read, but it's not necessary for the PR. Keeping a PR focused on a specific set of changes means we get nice, clean diffs.

**Comment by weavejester** [comment]
Fixed by 3b7020e6e6721fa4a431a79b164f132f22fdb3aa

**Pull request closed without merging**